django/db/backends/sqlite3/operations.py

```diff
@@ -370,8 +372,27 @@ def subtract_temporals(self, internal_type, lhs, rhs):
             return 'django_time_diff(%s, %s)' % (lhs_sql, rhs_sql), params
         return 'django_timestamp_diff(%s, %s)' % (lhs_sql, rhs_sql), params
 
-    def insert_statement(self, ignore_conflicts=False):
-        return 'INSERT OR IGNORE INTO' if ignore_conflicts else super().insert_statement(ignore_conflicts)
+    def insert_statement(self, on_conflict=None):
+        if on_conflict == OnConflict.Ignore:
+            return 'INSERT OR IGNORE INTO'
+
+        elif on_conflict == OnConflict.Update and not DatabaseFeatures.supports_update_conflicts_with_target:
+            return 'INSERT OR REPLACE INTO'
+
+        return super().insert_statement(on_conflict=on_conflict)
+
+    def on_conflict_suffix_sql(
+        self, opts, fields, on_conflict=None, update_fields=None, unique_fields=None
+    ):
+        if on_conflict == OnConflict.Update and DatabaseFeatures.supports_update_conflicts_with_target:
+            result = 'ON CONFLICT("%s") DO UPDATE SET ' % ('", "'.join(unique_fields))
+            result += ', '.join('%s=excluded.%s' % (field, field) for field in update_fields)
+            return result
```

Comment:
Use `self.quote_name()` to do the quoting of fields:

```suggestion
            return 'ON CONFLICT(%s) DO UPDATE SET %s' % (
                ', '.join(map(self.quote_name, unique_fields or ())),
                ', '.join(f'{field}=excluded.{field}' for field in map(self.quote_name, update_fields or ())),
            )
```